### PR TITLE
Normal Module replacement in falcon-client

### LINKED ISSUE #287: Overriding UI components
Research/investigate current issues

### PR DESCRIPTION
**Pull request by tutok**
resolve #287

### CODE REVIEW

**Comment by tutok** [comment]
How it works? Open `falcon-client.build.config.js`, then edit `moduleOverride` dictionary:

```
module.exports = {
  moduleOverride: {
    '@deity/falcon-ui/dist/components/Text': 'src/components/RedText'
  }
};
```

above configuration will tell falcon-client to use RedText component located in example project everywhere where falcon-ui's Text component is used, including other packages e.g. @deity/falcon-ecommerce-uikit

File: packages/falcon-dev-tools/normal-module-override-webpack-plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Normal Module override webpack plugin
+
+This webpack plugin allows overriding `module`s imported in third-party packages without modifying them.
+
+If you would like to find more information please check out the official documentation website [https://falcon.deity.io/](https://falcon.deity.io/)
```
Comment (Borales) [inline packages/falcon-dev-tools/normal-module-override-webpack-plugin/README.md]:
```suggestion
If you would like to find more information please check out the official documentation website [https://falcon.deity.io/docs/falcon-client/configurations#falcon-clientbuildconfigjs](https://falcon.deity.io/docs/falcon-client/configurations#falcon-clientbuildconfigjs)
```

File: packages/falcon-dev-tools/normal-module-override-webpack-plugin/index.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+const path = require('path');
+const glob = require('glob');
+
+module.exports = class NormalModuleOverridePlugin {
```
Comment (Borales) [inline packages/falcon-dev-tools/normal-module-override-webpack-plugin/index.js]: @tutok doesn't it require any base class from webpack? (looks like you're using some specific method names)

Comment (tutok) [reply packages/falcon-dev-tools/normal-module-override-webpack-plugin/index.js]: There is no somethig like base class for webpack plugins, more info here: https://webpack.js.org/api/plugins/

Comment (Borales) [reply packages/falcon-dev-tools/normal-module-override-webpack-plugin/index.js]: @tutok got it, thanks

**Comment by mateuszwozniak** [comment]
@tutok Can you please add relevant documentation with some examples to falcon-docs repo?

**Comment by mateuszwozniak** [comment]
Before merging add info to the changelog please.

**Pull request merged**